libr/util/diff.c

```diff
@@ -206,49 +214,28 @@ R_API bool r_diff_buffers_distance(RDiff *d, const ut8 *a, ut32 la, const ut8 *b
 			*similarity = 1.0;
 		return true;
 	}
-	totalsz = sizeof(int*) * (lb+1);
-	for(i = 0; i <= la; i++) {
-		totalsz += ((lb+1) * sizeof(int));
-	}
-	if (totalsz >= 1024 * 1024 * 1024) { // 1 GB of ram
-		char *szstr = r_num_units (NULL, totalsz);
-		eprintf ("Too much memory required (%s) to run distance diff, Use -c.\n", szstr);
-		free (szstr);
-		return false;
-	}
-	if ((m = malloc ((la+1) * sizeof(int*))) == NULL)
-		return false;
-	for(i = 0; i <= la; i++) {
-		if ((m[i] = malloc ((lb+1) * sizeof(int))) == NULL) {
-			eprintf ("Allocation failed\n");
-			while (i--)
-				free (m[i]);
-			free (m);
-			return false;
-		}
-	}
 
-	for (i = 0; i <= la; i++)
-		m[i][0] = i;
-	for (j = 0; j <= lb; j++)
-		m[0][j] = j;
+	for (i = 0; i < lb+1 ; i++) v0[i]=i;
+	
+	for (i = 0; i < la; i++) {
+		v1[0]=i + 1;
 
-	for (i = 1; i <= la; i++) {
-		for (j = 1; j <= lb; j++) {
-			int cost = (a[i-1] != b[j-1])? 1: 0;
-			tmin = R_MIN (m[i-1][j] + 1, m[i][j-1] + 1);
-			m[i][j] = R_MIN (tmin, m[i-1][j-1] + cost);
+		for (j = 0; j < lb; j++) {
+			int cost = (a[i]==b[j]) ? 0 : 1;
+			int smallest=((v1[j]+1) < (v0[j+1]+1)) ? v1[j]+1 : v0[j+1]+1;
+			smallest=(smallest < (v0[j]+cost)) ? smallest : v0[j]+cost;
+			v1[j+1]=smallest;
 		}
+
+		for (j = 0; j < lb+1; j++) v0[j]=v1[j];
```

Comment:
This is what I'm trying to avoid.. A wasteful copy from v1 to v0, instead just change the references.